postgrest/postgrest-postgis-spatial-api.md

```diff
@@ -0,0 +1,403 @@
+```
+
+It's important to mention here that you can call a function that takes a single parameter of type json by sending the header `Prefer: params=single-object` with your request. 
+That way the JSON request body will be used as the single argument.
+
+
+## Step 3 - Calculating the Length of a LineString
+
+Fantastic, now we can start with some more fun stuff.
+Let's go ahead and write a function which is able to calculate the length of a linestring provided in [EPSG:4326](https://spatialreference.org/ref/epsg/wgs-84/).
+Note, that the GeoJSON we are providing is in degrees, so we will have to transform to pseudo-mercator (or something similar).
+Additionally, we will cast it to `numeric` and after dividing it by 1.000 (to obtain a result in `kilometers`) we will round it to 2 decimals.
+
+
+```sh
+CREATE OR REPLACE FUNCTION api.calc_length(linestring json) RETURNS numeric AS $$ 
+
+    SELECT ROUND(
+      CAST(
+        ST_Length(
+          ST_Transform(
+            ST_GeomFromGeoJSON(linestring)
+          , 3857)
+        )/1000 AS numeric
+      ),2
+    )
+
+$$ LANGUAGE SQL;
+```
+
+Easy enough.
+Let's give this a try.
+
+
+```sh
+curl -X POST \
+  http://localhost:3000/rpc/calc_length \
+  -H 'Content-Type: application/json' \
+  -H 'Prefer: params=single-object' \
+  -d '{"type":"LineString","coordinates":[[-101.744384,39.32155],[-101.552124,39.330048],[-101.403808,39.330048],[-101.332397,39.364032],[-101.041259,39.368279],[-100.975341,39.304549],[-100.914916,39.245016],[-100.843505,39.164141],[-100.805053,39.104488],[-100.491943,39.100226],[-100.437011,39.095962],[-100.338134,39.095962],[-100.195312,39.027718],[-100.008544,39.010647],[-99.865722,39.00211],[-99.684448,38.972221],[-99.51416,38.929502],[-99.382324,38.920955],[-99.321899,38.895308],[-99.113159,38.869651],[-99.0802,38.85682],[-98.822021,38.85682],[-98.448486,38.848264],[-98.206787,38.848264],[-98.020019,38.878204],[-97.635498,38.873928]],"crs":{"type":"name","properties":{"name":"EPSG:4326"}}}'
+```
+
+Voila!
+For our route through Kansas, USA it returns:
+
+```sh
+[
+    {
+        "calc_length": 485.39
+    }
+]
+```
+
+kilometers!
+
+
+## Step 4 - Calculating the Area of a Polygon
+
+By now you have most likely picked up the gist of how easy it is to implement custom functions.
+Similary to the linestring example we could also calculate the area of a polygon.
+Let's make this a little trickier and implement it in a way enabling the user to send a `GeoJSON FeatureCollection`.
+You will notice that we have to extract the geometry from the feature collection at position 0, so the first element in the list which is our polygon.
```

Comment:
Wouldn't it be nice to return a list of areas for a list of features in the `FeatureCollection`? If it's not bloating the code too much.

Reply:
Yea, it would introduce more code, would like to keep it simple for now